go-controller/pkg/node/default_node_network_controller.go

```diff
@@ -494,6 +558,16 @@ func (nc *DefaultNodeNetworkController) Start(ctx context.Context) error {
 	nodeAnnotator := kube.NewNodeAnnotator(nc.Kube, node.Name)
 	waiter := newStartupWaiter()
 
+	// Get resources from Device Plugin if present
+	if err := retrieveDevicePluginResources(); err != nil {
```

Comment:
The function is named `retrive...` I'm a bit surprised that it has side-effects. 

In addition, can we move the  "if present" part out? It seems to me that both functions use something similar

Reply:
I think maybe handle is a better word?

Yes, moving part of it out makes sense. Changes made.